packages/shim/src/shim.ts

```diff
@@ -39,9 +39,12 @@ export class Shim {
    */
   public pushScope(client?: any): void {
     const usedClient = client || this.getCurrentClient();
+    // We want to clone the last scope and not create a new one
+    const stack = this.getStack();
+    const parentScope = stack[stack.length - 1].scope;
```

Comment:
Can we add an assertion here or even a fallback so we don't blow up in case stack is empty?